core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -701,7 +701,7 @@ private[spark] object Utils extends Logging {
     }
   }
 
-  private var customHostname: Option[String] = None
+  private var customHostname: Option[String] = sys.env.get("SPARK_LOCAL_HOSTNAME")
```

Comment:
There is already an environment variable called `SPARK_PUBLIC_DNS` in the docs. This is used to override the default host name in some cases (however, confusingly, in a smaller subset of cases). I wonder if we should just fall back to SPARK_PUBLIC_DNS here and expand its scope slightly. We'd need to audit all of the cases where this is used, but that might be preferable to introducing another override. We could also just have two overrides here, and we explain that SPARK_PUBLIC_DNS takes precedence and is only used in certain cases.